web/src/pages/billing/BillingCostByAnalysis.tsx

```diff
@@ -27,6 +29,36 @@ const BillingCostByAnalysis: React.FunctionComponent = () => {
 
     const [data, setData] = React.useState<any>(undefined)
 
+    const setArData = (arData: []) => {
+        setIsLoading(false)
+        // arData is an array of objects, we use only the first obejct
+        // in the future we maye have search by several ar_guids / author etc.
+        if (arData === undefined || arData.length === 0) {
+            // nothing found
+            setIsLoading(false)
+            return
+        }
+        const ar_record = arData[0]
+        if (
+            ar_record !== undefined &&
+            ar_record.total !== undefined &&
+            ar_record.total.ar_guid !== undefined
+        ) {
```

Comment:
Use optional chaining to simplify this:

```suggestion
        if (!!ar_record?.total?.ar_guid) {
```

Reply:
Thanks for the tip, have not thought about that :-)